blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
Do we really need to filter this out? It seems that `tensor.sqr(tensors).norm(2)` would work as well, and that's short enough to not warrant a separate function.

If we do want to start a collection of Theano convenience expressions, I would call them `blocks.expressions` and not `blocks.snippets`.